dali/c_api/c_api.cc

```diff
@@ -368,8 +368,16 @@ size_t daliMaxDimTensors(daliPipelineHandle* pipe_handle, int n) {
 }
 
 unsigned daliGetNumOutput(daliPipelineHandle* pipe_handle) {
-  dali::DeviceWorkspace* ws = reinterpret_cast<dali::DeviceWorkspace*>(pipe_handle->ws);
-  return ws->NumOutput();
+  dali::Pipeline *pipeline = reinterpret_cast<dali::Pipeline *>(pipe_handle->pipe);
+  return pipeline->num_outputs();
+}
```

Comment:
I've changed the implementation, because before it was possible to call this function before build and it returned 0 without errors.